validator/rpc/standard_api.go

```
@@ -382,3 +388,117 @@ func groupDeleteRemoteKeysErrors(req *ethpbservice.DeleteRemoteKeysRequest, erro
+	if s.validatorService == nil {
+		return nil, status.Error(codes.FailedPrecondition, "Validator service not ready.")
+	}
+	validatorKey := req.Pubkey
+	if len(validatorKey) != fieldparams.BLSPubkeyLength {
+		return nil, status.Errorf(
+			codes.InvalidArgument, "provided public key in path is not a valid bls public key, please check for correct hex format starting with 0x")
+	}
+	defaultOption := validatorServiceConfig.DefaultProposerOption()
+	encoded := hexutil.Encode(req.Ethaddress)
+	if !common.IsHexAddress(encoded) {
+		return nil, status.Errorf(
+			codes.InvalidArgument, "fee recipient is not a valid Ethereum address")
+	}
+	pOption := validatorServiceConfig.DefaultProposerOption()
+	pOption.FeeRecipient = common.BytesToAddress(req.Ethaddress)
+	switch {
+	case s.validatorService.ProposerSettings == nil:
+		s.validatorService.ProposerSettings = &validatorServiceConfig.ProposerSettings{
+			ProposeConfig: map[[fieldparams.BLSPubkeyLength]byte]*validatorServiceConfig.ProposerOption{
+				bytesutil.ToBytes48(validatorKey): &pOption,
+			},
+			DefaultConfig: &defaultOption,
+		}
+	case s.validatorService.ProposerSettings.ProposeConfig == nil:
+		s.validatorService.ProposerSettings.ProposeConfig = map[[fieldparams.BLSPubkeyLength]byte]*validatorServiceConfig.ProposerOption{
+			bytesutil.ToBytes48(validatorKey): &pOption,
+		}
+	default:
+		proposerOption, found := s.validatorService.ProposerSettings.ProposeConfig[bytesutil.ToBytes48(validatorKey)]
+		if found {
+			proposerOption.FeeRecipient = common.BytesToAddress(req.Ethaddress)
+		} else {
+			s.validatorService.ProposerSettings.ProposeConfig[bytesutil.ToBytes48(validatorKey)] = &pOption
+		}
+	}
+	// override the 200 success with 202 according to the specs
+	if err := grpc.SetHeader(ctx, metadata.Pairs("x-http-code", "202")); err != nil {
+		return &empty.Empty{}, fmt.Errorf("could not set custom success code header: %w", err)
+	}
```

I don't consider this an error condition. But if you do want to return an error here, then you should use `status.Errorf` and start with an uppercase.